protocol/0075-PLUP_protocol_upgrades.md

```diff
@@ -0,0 +1,166 @@
+- A synchronous restart is required
+- All node need to be restarted in a very short time so all state can be restore from ethereum, and the network can start properly with a checkpoint.
+
+Limited Network Life is not the end goal. This spec outlines how the protocol evolves from LNL checkpoints to rolling software updates, controlled by a reasonable set of governance and user controls.
+
+### How other protocols proceed
+
+Other protocol e.g Ethereum carry out updates in an asynchronous manner. Usually a new version of the protocol is made available, a hardcoded block height is set
+at which a new code path will be enabled. If enough node runners have deployed the updated code, then the network will continue with the new code path. Others nodes
+which haven't updated will then fork the blockchain.
+
+Cosmos is using a small program called cosmovisor. The cosmovisor is listening to proposals and when a proposal for upgrade is accepted, it is preparing the binary for a new release and handling the upgrade - https://docs.cosmos.network/main/run-node/cosmovisor.html.
+
+The idea in this spec is to draw inspiration from the design of cosmovisor and build vega-visor to manage protocol upgrades. 
+
+### Prior example of updgrading the nodes asynchronously
+
+Back in December 2021, vega proceeded to a LNL restore. Unfortunately, a bug in the code prevented the dispatch of the network parameters after the restore. This left
+the network in a semi invalid state where the network parameters were defined by the ones from the genesis block instead of the one from the checkpoint from the previous network.
+
+The solution employed at the time was to:
+- implement a hotfix
+- keep the hotfix behind a guard until a given time, once the time is reached network parameters would be dispatched.
+- distribute the code to the validator so they can test it.
+- then decide of an actual date which would give a week for the validator to asynchronously update their node.
+
+This upgrade went quite smoothly without any incident, although it was possible as the fix was located in a single place, and wasn't impacting much of the state
+(meaning no changes in the protobuf files were required for example.)
+
+## Protocol upgrades mechanism
+
+One of the main challenge with upgrading the network both asynchronously and without a restart / downtime is
+the management of the state, and possible incompatible changes between the state of two version of the protocol.
+
+The following describes the general workflow of upgrades: 
+1. Vega continuesly make new releases available 
+2. Validators can suggest to the network an upgrade with a given release tag and a block height for the upgrade to take place
+3. If a proposal gets enough votes (`protocol.upgrade.proposal.requiredMajority` of the validators, not stake) at the given height vega will take a snapshot automatically and stop processing further blocks until restarted by the vega-visor, the vega process manager. 
```

Comment:
I am guessing this was discussed - but does it make sense to count validators not something like "own + delegated stake" or "tendermint weight"? Because the purpose is for the chain to continue post upgrade - so I'd think we need not only majority but the "right ones". Thoughts?

Reply:
@barnabee - please can you review this and confirm if we should leave as is, or factor in the tendermint weight?

Reply:
@jgsbennett - once we have a decision here can you please add an AC accordingly?

Reply:
Decision in spec review meeting: We will remain as is and use count and not factor in score/weights for this milestone

Reply:
Net param is actually validators.vote.required . Rename here and in any ACs that mention.